### PR TITLE
gets NutritionDisplay use case working (except any updates to SearchRecipeView)

### LINKED ISSUE #13: nutrition stats - get api working
get nutrition api to work (so figure out http request format, etc.), implement as much as possible up to and maybe including view, but might not be possible until View code is reorganized

### PR DESCRIPTION
**Pull request by parkhangorodsky**
the Nutrition API works correctly, and errors are caught and dealt with. 

When an error 555 occurs, an empty Nutrition is made. Thus, later, in the SearchRecipeView, if an empty Nutrition is encountered, instead of nutritional info it should instead write something like "could not retrieve nutritional information for this recipe"

This merge will make no changes to the SearchRecipeView - this will be dealt with in another issue shortly (ASAP), this only ensures that for each recipe, the nutrition api is called, and that information is passed along, creating NutritionDisplayOutputData, and ending by calling SearchRecipePresenter. The next issue will resume from there.

### CODE REVIEW

File: src/main/java/frameworks/api/NutritionDisplayApi.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            nutritionJSONObject.put("displayMessage", "could not retrieve nutritional information for this recipe");
+            return nutritionJSONObject;
+        } else {
+            System.out.println("Request failed with code: " + response.code());
+            System.out.println("Response message: " + response.message());
+            throw new HttpResponseException("HTTP error code: " + response.code() + ", message: " + response.message() + " with URL: " + endpoint);
+        }
+    }
+
+    private class HttpResponseException extends RuntimeException {
```
Comment (seonghyunban) [inline src/main/java/frameworks/api/NutritionDisplayApi.java:87]: Abstraction

Comment (parkhangorodsky) [reply src/main/java/frameworks/api/NutritionDisplayApi.java:87]: agree!

File: src/main/java/use_cases/_common/xtra/json_processor/NutritionJSONHandler.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import entity.Nutrition;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface NutritionJSONHandler extends JSONNullHandler {
+
+     default List<Nutrition> convertJSONtoNutritionList(JSONObject nutritionJSONObject) {
```
Comment (seonghyunban) [inline src/main/java/use_cases/_common/xtra/json_processor/NutritionJSONHandler.java:12]: List<String> as a parameter?

File: src/main/java/use_cases/nutrition_display/use_case/input_data/NutritionDisplayInputData.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import use_cases.nutrition_display.use_case.interactor.NutritionDisplayInputBoundary;
+import use_cases.nutrition_display.use_case.interactor.NutritionDisplayInteractor;
+
+import java.util.List;
+
+public class NutritionDisplayInputData {
+    private String title;
+    private List<Ingredient> ingredients;
+
+    public NutritionDisplayInputData(List<Ingredient> ingredients) {
```
Comment (seonghyunban) [inline src/main/java/use_cases/nutrition_display/use_case/input_data/NutritionDisplayInputData.java]: Remove single parameter constructor

**Review by seonghyunban — COMMENTED** [review]
Looks fine overall. Check the comments!

**Pull request merged**